vars/protecodeExecuteScan.groovy

```diff
@@ -14,32 +14,29 @@ import static com.sap.piper.Prerequisites.checkScript
  */
 void call(Map parameters = [:]) {
     handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters,  failOnError: true) {
-
         def script = checkScript(this, parameters) ?: this
-
-        Map config
         def utils = parameters.juStabUtils ?: new Utils()
+        parameters.juStabUtils = null
         def jenkinsUtils = parameters.jenkinsUtilsStub ?: new JenkinsUtils()
+        parameters.jenkinsUtilsStub = null
 
         new PiperGoUtils(this, utils).unstashPiperBin()
         utils.unstash('pipelineConfigAndTests')
 
-        writeFile(file: METADATA_FILE, text: libraryResource(METADATA_FILE))
+        writeFile(file: ".pipeline/tmp/${METADATA_FILE}", text: libraryResource(METADATA_FILE))
 
         withEnv([
-            "PIPER_parametersJSON=${groovy.json.JsonOutput.toJson(parameters)}",
+            "PIPER_parametersJSON=${getParametersJSON(parameters)}",
         ]) {
-
             // get context configuration
-            config = readJSON (text: sh(returnStdout: true, script: "./piper getConfig --contextConfig --stepMetadata '${METADATA_FILE}'"))
+            Map config = readJSON (text: sh(returnStdout: true, script: "./piper getConfig --contextConfig --stepMetadata '.pipeline/tmp/${METADATA_FILE}'"))
 
             def creds = []
             if (config.protecodeCredentialsId) creds.add(usernamePassword(credentialsId: config.protecodeCredentialsId, passwordVariable: 'PIPER_password', usernameVariable: 'PIPER_user'))
             if (config.dockerCredentialsId) creds.add(file(credentialsId: config.dockerCredentialsId, variable: 'FILE_PATH'))
 
             // execute step
             withCredentials(creds) {
-
                 if(config.dockerCredentialsId) {
                     if (FILE_PATH) {
```

Comment:
@redehnroV pointing to the folder that contains the key file prevents me here from migrating to `piperExecuteBin`. Do you if passing the credentials file also work for this case?

Reply:
from `withCredentials`

Reply:
I still don't get it. Is `if (FILE_PATH)` synonymous to `if (env.FILE_PATH)`?!

Reply:
I am obviously missing some crucial knowledge about Groovy and/or Jenkins magic. I tested this in the Sandbox pipeline, and it resolves. I've looked into the documentation of `withCredentials`, but it doesn't mention this particular feature. If you can share a link where I can learn more, that would be greatly appreciated, I don't even know what to google for.

Reply:
> I still don't get it. Is `if (FILE_PATH)` synonymous to `if (env.FILE_PATH)`?!

Actually, I don't know. Interesting.. I remove this with https://github.com/SAP/jenkins-library/pull/1502